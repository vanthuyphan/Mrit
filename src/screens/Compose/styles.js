export default {
  containerImage: {
    flex: 1,
    width: null,
    height: null
  },
  box: {
    borderRadius: 5
  }
};
