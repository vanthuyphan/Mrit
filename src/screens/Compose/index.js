import React, {Component} from "react";
import {ImageBackground} from "react-native";
import {
  Container,
  Header,
  Title,
  Content,
  Text,
  Button,
  Icon,
  Right,
  Left,
  Body,
  Card,
  CardItem
} from "native-base";
import styles from "./styles";

const commonColor = require("../../theme/variables/commonColor");
const glow2 = require("../../../assets/glow2.png");

class Compose extends Component {
  render() {
    const navigation = this.props.navigation;
    return (
      <Container>
        <ImageBackground source={glow2} style={styles.containerImage}>
          <Header>
            <Left>
              <Button
                transparent
                onPress={() => navigation.navigate("DrawerOpen")}
              >
                <Icon active name="menu" />
              </Button>
            </Left>
            <Body>
              <Title>Compose</Title>
            </Body>
            <Right />
          </Header>

          <Content padder>
            <Card style={styles.box}>
              <CardItem
                style={{borderBottomWidth: 1, borderBottomColor: "#ddd"}}
              >
                <Text style={{color: commonColor.lightTextColor}}>From</Text>
                <Text
                  style={{
                    color: commonColor.inverseTextColor,
                    paddingLeft: 10
                  }}
                >
                  sanket@gmail.com
                </Text>
              </CardItem>

              <CardItem
                style={{borderBottomWidth: 1, borderBottomColor: "#ddd"}}
              >
                <Text style={{color: commonColor.lightTextColor}}>To</Text>
                <Text
                  style={{
                    color: commonColor.inverseTextColor,
                    paddingLeft: 10
                  }}
                >
                  pratik@gmail.com
                </Text>
              </CardItem>

              <CardItem
                style={{borderBottomWidth: 1, borderBottomColor: "#ddd"}}
              >
                <Text style={{color: commonColor.lightTextColor}}>Subject</Text>
                <Text
                  style={{
                    color: commonColor.inverseTextColor,
                    paddingLeft: 10
                  }}
                >
                  Enquiry about StrapMobile
                </Text>
              </CardItem>

              <CardItem>
                <Text style={{color: commonColor.inverseTextColor}}>
                  I watched Leicester City lose in the 1969 FA Cup final with my
                  dad when I was eight and cried all the way home. I have seen
                  them get promoted and relegated. I played eight years for
                  them. I even got a group of likeâ€‘minded fans and friends to
                  stump up a few quid to salvage the club when they went into
                  liquidation.
                </Text>
              </CardItem>

              <CardItem header style={styles.box}>
                <Button rounded style={{paddingHorizontal: 15}}>
                  <Text style={{color: commonColor.textColor}}>Send</Text>
                </Button>
              </CardItem>
            </Card>
          </Content>
        </ImageBackground>
      </Container>
    );
  }
}

export default Compose;
