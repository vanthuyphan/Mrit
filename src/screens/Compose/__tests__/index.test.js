import "react-native";
import React from "react";
import Compose from "../index";
// Note: test renderer must be required after react-native.
import renderer from "react-test-renderer";

it("renders correctly", () => {
  const tree = renderer.create(<Compose />).toJSON();
  expect(tree).toMatchSnapshot();
});
