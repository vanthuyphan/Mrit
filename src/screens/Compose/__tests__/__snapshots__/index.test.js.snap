// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "flex": 1,
      "height": 1334,
    }
  }
  virtual={undefined}
>
  <Image
    source={1}
    style={
      Object {
        "flex": 1,
        "height": null,
        "width": null,
      }
    }
  >
    <View>
      <View
        style={
          Object {
            "backgroundColor": "#F8F8F8",
            "borderBottomColor": "#a7a6ab",
            "borderBottomWidth": 0.5,
            "elevation": 3,
            "flexDirection": "row",
            "height": 64,
            "justifyContent": "center",
            "left": 0,
            "paddingLeft": 10,
            "paddingRight": 10,
            "paddingTop": 15,
            "right": 0,
            "shadowColor": undefined,
            "shadowOffset": undefined,
            "shadowOpacity": undefined,
            "shadowRadius": undefined,
            "top": 0,
          }
        }
        virtual={undefined}
      >
        <View
          style={
            Object {
              "alignItems": "flex-start",
              "alignSelf": "center",
              "flex": 1,
            }
          }
          virtual={undefined}
        >
          <View
            accessibilityComponentType={undefined}
            accessibilityLabel={undefined}
            accessibilityTraits={undefined}
            accessible={true}
            collapsable={undefined}
            hitSlop={undefined}
            isTVSelectable={true}
            nativeID={undefined}
            onLayout={undefined}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "alignItems": "center",
                "alignSelf": null,
                "backgroundColor": "transparent",
                "borderBottomWidth": null,
                "borderColor": null,
                "borderLeftWidth": null,
                "borderRadius": 5,
                "borderRightWidth": null,
                "borderTopWidth": null,
                "elevation": 0,
                "flexDirection": "row",
                "height": 45,
                "justifyContent": "space-between",
                "marginLeft": -3,
                "opacity": 1,
                "paddingBottom": 6,
                "paddingLeft": 6,
                "paddingRight": 6,
                "paddingTop": 6,
                "shadowColor": null,
                "shadowOffset": null,
                "shadowOpacity": null,
                "shadowRadius": null,
              }
            }
            testID={undefined}
            tvParallaxProperties={undefined}
          >
            <Text
              accessible={true}
              active={true}
              allowFontScaling={false}
              disabled={false}
              ellipsizeMode="tail"
              style={
                Array [
                  Object {
                    "color": undefined,
                    "fontSize": 12,
                  },
                  Array [
                    Object {
                      "color": "#000",
                      "fontSize": 30,
                    },
                    Object {
                      "color": "#007aff",
                      "fontSize": 33,
                      "marginLeft": 2,
                      "marginRight": 2,
                      "marginTop": 2,
                      "paddingTop": 2,
                    },
                  ],
                  Object {
                    "fontFamily": "Ionicons",
                    "fontStyle": "normal",
                    "fontWeight": "normal",
                  },
                ]
              }
              virtual={undefined}
            >
              ï‡ƒ
            </Text>
          </View>
        </View>
        <View
          style={
            Object {
              "alignItems": "center",
              "alignSelf": "center",
              "flex": 1,
            }
          }
          virtual={undefined}
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            disabled={false}
            ellipsizeMode="tail"
            numberOfLines={1}
            style={
              Object {
                "color": "#000",
                "fontFamily": "System",
                "fontSize": 17,
                "fontWeight": "600",
                "textAlign": "center",
              }
            }
            virtual={undefined}
          >
            Compose
          </Text>
        </View>
        <View
          style={
            Object {
              "alignItems": "flex-end",
              "alignSelf": "center",
              "flex": 1,
              "flexDirection": "row",
              "justifyContent": "flex-end",
            }
          }
          virtual={undefined}
        />
      </View>
    </View>
    <RCTScrollView
      automaticallyAdjustContentInsets={false}
      contentInset={
        Object {
          "bottom": 0,
        }
      }
      enableAutoAutomaticScroll={true}
      enableResetScrollToCoords={true}
      extraHeight={75}
      extraScrollHeight={0}
      keyboardDismissMode="interactive"
      keyboardOpeningTime={250}
      onScroll={[Function]}
      padder={true}
      resetScrollToCoords={
        Object {
          "x": 0,
          "y": 0,
        }
      }
      scrollEventThrottle={0}
      showsVerticalScrollIndicator={true}
      style={
        Object {
          "backgroundColor": "transparent",
          "flex": 1,
          "paddingBottom": 10,
          "paddingLeft": 10,
          "paddingRight": 10,
          "paddingTop": 10,
        }
      }
      virtual={undefined}
    >
      <View>
        <View
          style={
            Array [
              Object {
                "backgroundColor": "#fff",
                "borderBottomWidth": 0.5,
                "borderColor": "#ccc",
                "borderLeftWidth": 0.5,
                "borderRadius": 2,
                "borderRightWidth": 0.5,
                "borderTopWidth": 0.5,
                "elevation": 3,
                "flex": 1,
                "flexWrap": "wrap",
                "marginBottom": 5,
                "marginLeft": 2,
                "marginRight": 2,
                "marginTop": 5,
                "shadowColor": "#000",
                "shadowOffset": Object {
                  "height": 2,
                  "width": 0,
                },
                "shadowOpacity": 0.1,
                "shadowRadius": 1.5,
              },
              Object {
                "borderRadius": 5,
              },
            ]
          }
          virtual={undefined}
        >
          <View
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "backgroundColor": "#fff",
                  "borderRadius": 2,
                  "flexDirection": "row",
                  "paddingBottom": 10,
                  "paddingLeft": 15,
                  "paddingRight": 15,
                  "paddingTop": 10,
                },
                Object {
                  "borderBottomColor": "#ddd",
                  "borderBottomWidth": 1,
                },
              ]
            }
            virtual={undefined}
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              disabled={false}
              ellipsizeMode="tail"
              style={
                Array [
                  Object {
                    "color": "#000",
                    "fontFamily": "System",
                    "fontSize": 16,
                  },
                  Object {
                    "color": "#999",
                  },
                ]
              }
              uppercase={false}
              virtual={undefined}
            >
              From
            </Text>
            <Text
              accessible={true}
              allowFontScaling={true}
              disabled={false}
              ellipsizeMode="tail"
              style={
                Array [
                  Object {
                    "color": "#000",
                    "fontFamily": "System",
                    "fontSize": 16,
                  },
                  Object {
                    "color": "#000",
                    "paddingLeft": 10,
                  },
                ]
              }
              uppercase={false}
              virtual={undefined}
            >
              sanket@gmail.com
            </Text>
          </View>
          <View
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "backgroundColor": "#fff",
                  "borderRadius": 2,
                  "flexDirection": "row",
                  "paddingBottom": 10,
                  "paddingLeft": 15,
                  "paddingRight": 15,
                  "paddingTop": 10,
                },
                Object {
                  "borderBottomColor": "#ddd",
                  "borderBottomWidth": 1,
                },
              ]
            }
            virtual={undefined}
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              disabled={false}
              ellipsizeMode="tail"
              style={
                Array [
                  Object {
                    "color": "#000",
                    "fontFamily": "System",
                    "fontSize": 16,
                  },
                  Object {
                    "color": "#999",
                  },
                ]
              }
              uppercase={false}
              virtual={undefined}
            >
              To
            </Text>
            <Text
              accessible={true}
              allowFontScaling={true}
              disabled={false}
              ellipsizeMode="tail"
              style={
                Array [
                  Object {
                    "color": "#000",
                    "fontFamily": "System",
                    "fontSize": 16,
                  },
                  Object {
                    "color": "#000",
                    "paddingLeft": 10,
                  },
                ]
              }
              uppercase={false}
              virtual={undefined}
            >
              pratik@gmail.com
            </Text>
          </View>
          <View
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "backgroundColor": "#fff",
                  "borderRadius": 2,
                  "flexDirection": "row",
                  "paddingBottom": 10,
                  "paddingLeft": 15,
                  "paddingRight": 15,
                  "paddingTop": 10,
                },
                Object {
                  "borderBottomColor": "#ddd",
                  "borderBottomWidth": 1,
                },
              ]
            }
            virtual={undefined}
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              disabled={false}
              ellipsizeMode="tail"
              style={
                Array [
                  Object {
                    "color": "#000",
                    "fontFamily": "System",
                    "fontSize": 16,
                  },
                  Object {
                    "color": "#999",
                  },
                ]
              }
              uppercase={false}
              virtual={undefined}
            >
              Subject
            </Text>
            <Text
              accessible={true}
              allowFontScaling={true}
              disabled={false}
              ellipsizeMode="tail"
              style={
                Array [
                  Object {
                    "color": "#000",
                    "fontFamily": "System",
                    "fontSize": 16,
                  },
                  Object {
                    "color": "#000",
                    "paddingLeft": 10,
                  },
                ]
              }
              uppercase={false}
              virtual={undefined}
            >
              Enquiry about StrapMobile
            </Text>
          </View>
          <View
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "#fff",
                "borderRadius": 2,
                "flexDirection": "row",
                "paddingBottom": 10,
                "paddingLeft": 15,
                "paddingRight": 15,
                "paddingTop": 10,
              }
            }
            virtual={undefined}
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              disabled={false}
              ellipsizeMode="tail"
              style={
                Array [
                  Object {
                    "color": "#000",
                    "fontFamily": "System",
                    "fontSize": 16,
                  },
                  Object {
                    "color": "#000",
                  },
                ]
              }
              uppercase={false}
              virtual={undefined}
            >
              I watched Leicester City lose in the 1969 FA Cup final with my dad when I was eight and cried all the way home. I have seen them get promoted and relegated. I played eight years for them. I even got a group of likeâ€‘minded fans and friends to stump up a few quid to salvage the club when they went into liquidation.
            </Text>
          </View>
          <View
            header={true}
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "backgroundColor": "#fff",
                  "borderBottomWidth": null,
                  "borderRadius": 2,
                  "flexDirection": "row",
                  "paddingBottom": 15,
                  "paddingLeft": 15,
                  "paddingRight": 15,
                  "paddingTop": 15,
                },
                Object {
                  "borderRadius": 5,
                },
              ]
            }
            virtual={undefined}
          >
            <View
              accessibilityComponentType={undefined}
              accessibilityLabel={undefined}
              accessibilityTraits={undefined}
              accessible={true}
              collapsable={undefined}
              hitSlop={undefined}
              isTVSelectable={true}
              nativeID={undefined}
              onLayout={undefined}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "alignItems": "center",
                  "alignSelf": "flex-start",
                  "backgroundColor": "#007aff",
                  "borderBottomWidth": null,
                  "borderColor": "#007aff",
                  "borderLeftWidth": null,
                  "borderRadius": 57,
                  "borderRightWidth": null,
                  "borderTopWidth": null,
                  "elevation": 2,
                  "flexDirection": "row",
                  "height": 45,
                  "justifyContent": "space-between",
                  "opacity": 1,
                  "paddingBottom": 6,
                  "paddingHorizontal": 15,
                  "paddingLeft": 26,
                  "paddingRight": 26,
                  "paddingTop": 6,
                }
              }
              testID={undefined}
              tvParallaxProperties={undefined}
            >
              <Text
                accessible={true}
                allowFontScaling={true}
                disabled={false}
                ellipsizeMode="tail"
                style={
                  Array [
                    Object {
                      "color": "#fff",
                      "fontFamily": "System",
                      "fontSize": 16.5,
                      "lineHeight": 19,
                      "marginLeft": 0,
                      "marginRight": 0,
                    },
                    Object {
                      "color": "#fff",
                    },
                  ]
                }
                uppercase={false}
                virtual={undefined}
              >
                Send
              </Text>
            </View>
          </View>
        </View>
      </View>
    </RCTScrollView>
  </Image>
</View>
`;
